### PR TITLE
Update to JUnit5, Update to J17 and J21 for tests

### PR DESCRIPTION
**Pull request by NotStirred**
I switched from google truth to assertj, as truth depends heavily on junit4. No code changes thankfully, only imports (apis we use are identical)

Only thing of note: `QuickMathTest#testAbs_1` was previously incorrect (and now fails, relevant part is commented out)

Probably best not to squash as the two commits are entirely separate

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1,2 +1,6 @@
+plugins {
```
Comment (leMaik) [inline settings.gradle:1]: What does this plugin do? Maybe add a comment

**Review by leMaik — CHANGES_REQUESTED** [review]

File: settings.gradle (hunk 1 above)
Comment (NotStirred) [reply settings.gradle:1]: Ah this is the default toolchain resolver of gradle see [the docs](https://docs.gradle.org/current/userguide/toolchains.html#sub:download_repositories). Will add comment if you'd like

**Comment by NotStirred** [comment]
I moved most things into the subproject closure so it should be much easier to understand. Everything now has junit tests and jacoco coverage enabled.

**Comment by leMaik** [comment]
Maybe we can keep the launcher compatible with Java 8, just in case.

**Review by leMaik — APPROVED** [review]
Nice! Next step is increasing the test coverage.

**Pull request merged**